test/conftest.py

```diff
@@ -45,6 +45,40 @@ def scipy_network():
     return n
 
 
+@pytest.fixture(scope="function")
+def geo_components_network():
+    # Constructing a basic network using all 'geo_components'.
+    # geo_components = {"Bus", "Line", "Link", "Transformer"}
+    network = pypsa.Network()
+    num_components = 5
+
+    for i in range(num_components):
+        network.add("Bus", f"bus_{i}")
+
```

Comment:
Can you not avoid this and use of of the other networks e.g. ac_dc_network or scipy_network?

Reply:
Implementing this to streamline testing and eliminate the need to repeatedly add geo components. This will also simplify the addition of future Geopandas-related tests.